src/library_wasmfs_opfs.js

```diff
@@ -232,20 +301,30 @@ mergeInto(LibraryManager.library, {
       {{{ makeSetValue('errPtr', 0, 'err', 'i32') }}};
     }
     wasmfsOPFSAccessHandles.free(accessID);
-    _emscripten_proxy_finish(ctx);
+    wasmfsOPFSProxyFinish(ctx);
   },
 
+  _wasmfs_opfs_close_blob__sig: 'vi',
   _wasmfs_opfs_close_blob__deps: ['$wasmfsOPFSBlobs'],
   _wasmfs_opfs_close_blob: function(blobID) {
     wasmfsOPFSBlobs.free(blobID);
   },
 
+  _wasmfs_opfs_read_access__sig: 'iipii',
   _wasmfs_opfs_read_access__deps: ['$wasmfsOPFSAccessHandles'],
+#if USE_PTHREADS
   _wasmfs_opfs_read_access: function(accessID, bufPtr, len, pos) {
+#else
+  _wasmfs_opfs_read_access: async function(accessID, bufPtr, len, pos) {
+#endif
     let accessHandle = wasmfsOPFSAccessHandles.get(accessID);
     let data = HEAPU8.subarray(bufPtr, bufPtr + len);
     try {
+#if USE_PTHREADS
       return accessHandle.read(data, {at: pos});
+#else
+      return await accessHandle.read(data, {at: pos});
```

Comment:
Can we ifdef out only the `await` here? I also like your idea to add macros for this.